### PR TITLE
Replace old watchlist view code by the new_watchlist one in WebUI

### PR DESCRIPTION
**Pull request by saraycp**
We are replacing the old watchlist code with the new one step by step.

In this case, we focus on the WebUI and 
- remove the view code we used to render for the old watchlist in favour of the new watchlist one,
- remove checks of the feature flag in the WatchedItemsController.
- adapt tests

**How to test?**

Play around adding and removing items from the Watchlist.

### CODE REVIEW

**Comment by saraycp** [comment]
**DO NOT MERGE** until PR#13431 is merged and we rebase this PR.

**Review by danidoni — APPROVED** [review]

**Review by danidoni — APPROVED** [review]

**Pull request merged**
